base-template.sh

```diff
@@ -378,29 +395,30 @@ process_shared_hooks() {
 #####################################################
 update_shared_hooks_if_appropriate() {
     # run an init/update if we are after a "git pull" or triggered manually
-    if [ "$HOOK_NAME" = "post-merge" ] || [ "$HOOK_NAME" = ".githooks.shared.trigger" ]; then
+    if [ "$HOOK_NAME" = "post-merge" ] ||
+        [ "$HOOK_NAME" = "post-checkout" ] && [ "$1" = "$GIT_NULL_REF" ] ||
+        [ "$HOOK_NAME" = ".githooks.shared.trigger" ]; then
         # split on comma and newline
         IFS=",
         "
 
         for SHARED_REPO in $SHARED_REPOS_LIST; do
             mkdir -p ~/.githooks/shared
 
-            NORMALIZED_NAME=$(echo "$SHARED_REPO" |
-                sed -E "s#.*[:/](.+/.+)\\.git#\\1#" |
-                sed -E "s/[^a-zA-Z0-9]/_/g")
+            set_shared_root "$SHARED_REPO"
 
-            if [ -d ~/.githooks/shared/"$NORMALIZED_NAME"/.git ]; then
+            if [ -d "$SHARED_ROOT/.git" ]; then
                 echo "* Updating shared hooks from: $SHARED_REPO"
-                PULL_OUTPUT=$(cd ~/.githooks/shared/"$NORMALIZED_NAME" && git pull 2>&1)
+                PULL_OUTPUT=$(cd "$SHARED_ROOT" && git pull 2>&1)
                 # shellcheck disable=SC2181
                 if [ $? -ne 0 ]; then
                     echo "! Update failed, git pull output:"
                     echo "$PULL_OUTPUT"
                 fi
             else
                 echo "* Retrieving shared hooks from: $SHARED_REPO"
-                CLONE_OUTPUT=$(cd ~/.githooks/shared && git clone "$SHARED_REPO" "$NORMALIZED_NAME" 2>&1)
+                [ -d "$SHARED_ROOT" ] && rm -rf "$SHARED_ROOT"
```

Comment:
Out of curiosity, why delete the folder first? Would `git clone` fail if the directory exists?

Reply:
jep, git clone would fail, saying already existing...